### ISSUE TITLE
A way to render starting dates that are too early

### ISSUE THREAD

**Issue by levantis** [raphink/moderntimeline#4]
It would be great if when the starting date is earlier than the "firstyear", the bar is drawn as a gradient with the white end at "firstyear".

I tried to submit a patch, but it's taking me too much time to get a hang of tikz :(

**Comment by leezu** [comment]
In my opinion the gradient would make sense. In case you don't have the time I'd be very happy if you could shortly state your opinion about the feasibility and the right approach so I could try to look into it :)

**Comment by raphink** [comment]
I don't have the time indeed, but I'll welcome a PR for it.

**Comment by tristancollins** [comment]
I'm happy to share my modifications to the original code to give output like:

http://tex.stackexchange.com/a/206792/56813

I implemented gradients to show 'to present' roles (as seen in the top section - not a good example...), but this could be reversed to show roles starting before the timeline period.

I hoped to make my changes in to a PR, but never had the time...

Perhaps I'll have another look.

Tristan

**Comment by raphink** [comment]
Thanks, that looks good. A PR shouldn't take too long, especially if you already have a clone of the repository.

**Comment by tristancollins** [comment]
Yes - that's what I thought... And I made my own commands (with _TC) based on yours so should be a copy and paste job.

I think one issue I had was that I modified the .sty. Not the DTX file. And then I started playing with package options to allow switch on the alternative placements etc, so it would be a more elegant/correct solution.

I'll see if I can give it a go tomorrow. Could be opening up a can of worms!

Tristan

**Comment by raphink** [comment]
Hehe OK. At least if you integrate it in the DTX, it'll become official and maintained in future versions, so that's good for you, too.

**Comment by raphink** [comment]
FTR, https://tex.stackexchange.com/a/545111/951 suggests a way to implement this feature. I'd welcome a PR to add this properly to the package.
